doc/design/dist/README.md

```diff
@@ -0,0 +1,190 @@
+# Distributed Training Design Doc
+
+## Objective
+
+We want Paddle to support training on the general-purpose cluster. The cluster runs Paddle, the web server (e.g., Nginx), the log collector (e.g., fluentd), the distributed queue service (e.g., Kafka), the log joiner and other data processors written using Storm, Spark, and Hadoop MapReduce on the same cluster. As illustrated in the following graph:
+
+<img src="src/arch.png"/>
+
+This poses new challenges for Paddle,
+
+- Paddle need to be fault tolerant.
+- Input training data can be online data from real time logs or batch data from distributed file system.
+- User needs a simple way to train model on Paddle cloud. Complexities such as job scheduling should be hidden from user.
+
+## Training Job
+
+A training job will be created once user asks Paddle cloud to train a model. The training job is made up of different processes that collaboratively consume data and produce a trained model. There are three kinds of processes:
+
+- Master process
+- Trainer process
+- Parameter server process
+
+One training job will only have one master process, typically multiple trainer processes and parameter server processes. Their relation is illustrated in the following graph:
+
+<img src="src/paddle-model-sharding.png" height="400"/>
+
+### Master Process
+
+The master process will:
+
+- Do [parameter server selection](#parameter-server-selection).
+- Shard dataset into [tasks](#task) and dispatch tasks to trainers.
+- Keep track of training progress on the dataset with [task queue](#task-queue). A training job will iterate on the dataset for a full pass until it goes into next pass.
+
+Now we will explain the concepts mentioned above:
+
+#### Selection Request
+
+The selection request is a request that the master sends to a parameter server candidate, making it a parameter server available to the trainers. It contains information such as the parameter server index, the optimizaiton algorithm, the parameter save period, and the path for saving parameters.
+
+#### Parameter Server Selection
+
+The parameter server selection process selects parameter servers from parameter server candidates. It ensures the parameter servers that the trainers see are in consistent order, since the trainer needs to decide the parameter placement according to the consistent order.
+
+The selection process is as follows:
+
+- The master watches `/ps_candidate/` prefix in etcd. When a parameter server candidate joins and there is not enough parameter servers, the master will remove the candidate's entry in `/ps_candidate/` and send a [selection reqeust](#selection-request) to the candidate. Upon receiving the request, the candidate will set key `/ps/<index>` in etcd with a lease to make itself available for the trainers. The `<index>` is from the selection request.
```

Comment:
Removed the concept of parameter candidate. And added a key value pair in etcd `/ps_desired` for number of desired parameter servers.